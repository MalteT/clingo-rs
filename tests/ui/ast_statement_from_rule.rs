use clingo::*;

fn main() {
    let sym = Symbol::create_id("test", true).unwrap();
    let mut term1 = ast::Term::from(sym);
    let mut term2 = ast::Term::from(sym);
    let mut lit1 = ast::Literal::from_term(ast::Sign::None, &term1);
    let mut lit2 = ast::Literal::from_term(ast::Sign::None, &term2);
    let hlit = ast::HeadLiteral::from(&lit1);
    let blit = ast::BodyLiteral::from_literal(ast::Sign::None, &lit2);
    let body = vec![blit];
    let mut rule = ast::Rule::new(hlit, &body);
    let stmt = rule.ast_statement();
    
    term1 = ast::Term::from(sym);
    term2 = ast::Term::from(sym);
    lit1 = ast::Literal::from_term(ast::Sign::None, &term1);
    lit2 = ast::Literal::from_term(ast::Sign::None, &term2);
    rule = ast::Rule::new(hlit, &body);
    drop(body);
    let _end = (lit1, lit2, rule, stmt);
}