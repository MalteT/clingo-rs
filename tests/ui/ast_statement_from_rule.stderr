error[E0505]: cannot move out of `rule` because it is borrowed
  --> $DIR/ast_statement_from_rule.rs:12:10
   |
11 |     let stmt = rule.ast_statement();
   |                ---- borrow of `rule` occurs here
12 |     drop(rule);
   |          ^^^^ move out of `rule` occurs here
13 |     let _end = stmt;
   |                ---- borrow later used here

For more information about this error, try `rustc --explain E0505`.
